src/nodes/dir.rs

```diff
@@ -39,6 +41,99 @@ fn split_components<'a>(components: &'a [Component<'a>]) -> (&'a OsStr, &'a [Com
     (name, &components[1..])
 }
 
+/// Handle for an open directory.
+struct OpenDir {
+    // These are copies of the fields that also exist in the Dir corresponding to this OpenDir.
+    // Ideally we could just hold an immutable reference to the Dir instance... but this is hard
+    // to do because, when opendir() gets called on an abstract Node, we do not get access to the
+    // Arc<Node> that corresponds to it (to clone it).  Given that these values are immutable on
+    // the node, holding a copy here is fine.
+    inode: u64,
+    writable: bool,
+    state: Arc<Mutex<MutableDir>>,
+
+    /// Handle for the open directory file descriptor.  This is `None` if the directory does not
+    /// have an underlying path.
+    handle: Mutex<Option<rawdir::Dir>>,
+}
+
+impl Handle for OpenDir {
```

Comment:
Override other methods with EISDIR. It is very possible that the kernel does not type-check these things for you, so you need to guard against it.

Reply:
Same as above. Added a test.